README.md
# How to App Store
v 0.2 [Changelog](https://github.com/AlexBezuska/How-To-App-Store/blob/master/CHANGELOG.md)

A guide on how to release apps and games for the major platforms including tools for generating correct icon sets and screenshot sizes.

This is a community project intended to help developers focus on the fun part of game and app creation - actually working on the project itself!

## Guide:

- Platforms:
  - [Apple App Store guide](https://github.com/AlexBezuska/How-To-App-Store/blob/master/guide/Apple_App_Store.md)


- Scripts:
  - [Icon Generator Script](https://github.com/AlexBezuska/How-To-App-Store#how-to-use)


## We need your help!
### What works now
**Tested on Mac OS X** :
- Icon generation for web, iOS, and Windows phone 8

### How you can help
Please feel free to help with this project, specific needs and features wanted below in the `Feature Requests` and `Contributing` sections.


## Prerequisites & Installation

- Nothing needed for the [guide](https://github.com/AlexBezuska/How-To-App-Store#guide), it's just for reading, but there are a few steps if you want to use the scripts.

- Terminal
  - If you are using MacOS or Linux you already have this
  - No Windows instructions / build process created yet, [help wanted](https://github.com/AlexBezuska/How-To-App-Store#feature-requests)!


- [ImageMagick](https://www.imagemagick.org)
  - MacOS using [Homebrew](https://brew.sh/)
    - If you don't have Homebrew it can be installed with this command in your terminal: `/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`
    - Once you have Homebrew you can run `brew install ImageMagick`
  - Ubuntu
    - ```sudo apt-get install imagemagick```


- Clone this repository to your local machine(recommended)
  - ```git clone git@github.com:AlexBezuska/How-To-App-Store.git```
  - or download the zip (less recommended but it will work)


## How to use


 #### Generate app icons:
- Copy your app icon (2048 x 2048 px png format) into the `input` directory and name it `icon.png`
- In your terminal move into the root of the `How-To-App-Store` project ex. ``` cd How-To-App-Store ```
- run `./build.sh`
- A `output` directory will be created or updated with the output files in their respective subfolders


## Feature requests

- [Add installation & build process for Windows](https://github.com/AlexBezuska/How-To-App-Store/issues/1)
- [Add Google Play Store guide](https://github.com/AlexBezuska/How-To-App-Store/issues/2)
- [Add Google Play Store icon / screenshot dimensions](https://github.com/AlexBezuska/How-To-App-Store/issues/3)
- [Add Itch.io guide](https://github.com/AlexBezuska/How-To-App-Store/issues/4)
- [Add Itch.io screenshot and gif preview dimensions](https://github.com/AlexBezuska/How-To-App-Store/issues/5)
- [Add Steam guide](https://github.com/AlexBezuska/How-To-App-Store/issues/6)
- [Add Windows Store guide](https://github.com/AlexBezuska/How-To-App-Store/issues/7)
- [Add Windows Store icon / screenshot dimensions](https://github.com/AlexBezuska/How-To-App-Store/issues/8)
- [Add Mac App Store guide](https://github.com/AlexBezuska/How-To-App-Store/issues/9)
- [Add Mac App Store icon / screenshot dimensions](https://github.com/AlexBezuska/How-To-App-Store/issues/10)
- [Figure out base aspect ratios and max sizes for screen shots that a script can use to generate all the other sizes](https://github.com/AlexBezuska/How-To-App-Store/issues/11)
- Help add [github issues](https://github.com/AlexBezuska/How-To-App-Store/issues) for stores and platforms I am missing, data that is missing etc.



## Contributing

If you are interested in participating in this project, please feel free to send a bug or feature request by [submitting an issue](https://github.com/AlexBezuska/How-To-App-Store/issues), or submit a PR for me to review.

## Versioning

We use [SemVer](http://semver.org/) for versioning.

## Authors

* **[Alex Bezuska](https://github.com/AlexBezuska)** - *Initial work*

See also the list of [contributors](https://github.com/AlexBezuska/How-To-App-Store/contributors) who participated in this project.

## License

This project is licensed under the MIT License - see the [LICENSE.TXT](LICENSE.TXT) file for details


CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).

## [0.2] - 2017-10-10
- add apple app store guide

## [0.1] - 2017-10-10
### Added
- update folder structure
- add instructions to README.md
- update icons for iOS


square-icons.sh
#!/bin/sh
base=$1

mkdir web
# Apple touch icons
convert "$base" -resize "57x57" -unsharp 1x4 "web/icon-57x57.png"
convert "$base" -resize "60x60" -unsharp 1x4 "web/icon-60x60.png"
convert "$base" -resize "72x72" -unsharp 1x4 "web/icon-72x72.png"
convert "$base" -resize "76x76" -unsharp 1x4 "web/icon-76x76.png"
convert "$base" -resize "114x114" -unsharp 1x4 "web/icon-114x114.png"
convert "$base" -resize "120x120" -unsharp 1x4 "web/icon-120x120.png"
convert "$base" -resize "144x144" -unsharp 1x4 "web/icon-144x144.png"
convert "$base" -resize "152x152" -unsharp 1x4 "web/icon-152x152.png"
convert "$base" -resize "180x180" -unsharp 1x4 "web/icon-180x180.png"
#convert "$base" -resize "192x192" -unsharp 1x4 "web/apple-icon-precomposed.png"
#convert "$base" -resize "192x192" -unsharp 1x4 "web/apple-icon.png"
# Android touch icon
convert "$base" -resize "192x192"  -unsharp 1x4 "web/icon-192x192.png"
# favicons
convert "$base" -resize "16x16" -unsharp 1x4 -colors 256 "web/favicon.ico"
convert "$base" -resize "16x16" -unsharp 1x4 "web/icon-16x16.png"
convert "$base" -resize "32x32" -unsharp 1x4 "web/icon-32x32.png"
convert "$base" -resize "96x96" -unsharp 1x4 "web/icon-96x96.png"
convert "$base" -resize "128x128" -unsharp 1x4 "web/icon-128x128.png"



mkdir ios
# iPhone App Icon
convert "$base" -resize "120x120"-unsharp 1x4 "ios/Icon-60@2x.png"
# iPhone App Icon for Retina display
convert "$base" -resize "180x180"-unsharp 1x4 "ios/Icon-60@3x.png"
# iPad App Icon
convert "$base" -resize "76x76"-unsharp 1x4 "ios/Icon-76.png"
# iPad App Icon for Retina display
convert "$base" -resize "152x152"-unsharp 1x4 "ios/Icon-76@2x.png"
# iPad App Icon for Retina display
convert "$base" -resize "167x167"-unsharp 1x4 "ios/Icon-83.5@2x.png"
# App Icon large
convert "$base" -resize "1024x1024" -unsharp 1x4 "ios/Icon-1024.png"
# App Submission
convert "$base" -resize "2048x2048" -unsharp 1x4 "ios/Icon-2048.png"



mkdir windowsPhone8
# Windows phone 8.1 Square 71×71 Logo
convert "$base" -resize "170x170" -unsharp 1x4 "windowsPhone8/170×170Logo.scale-240.png"
convert "$base" -resize "99x99" -unsharp 1x4 "windowsPhone8/99×99Logo.scale-140.png"
convert "$base" -resize "71x71" -unsharp 1x4 "windowsPhone8/71×71Logo.scale-100.png"
# Windows phone 8.1 Square 150×150 Logo
convert "$base" -resize "360x360" -unsharp 1x4 "windowsPhone8/360×360Logo.scale-240.png"
convert "$base" -resize "210x210" -unsharp 1x4 "windowsPhone8/210×210Logo.scale-140.png"
convert "$base" -resize "150x150" -unsharp 1x4 "windowsPhone8/150×150Logo.scale-100.png"
# Windows phone 8.1 Square 44×44 Logo
convert "$base" -resize "106x106" -unsharp 1x4 "windowsPhone8/106×106Logo.scale-240.png"
convert "$base" -resize "62x62" -unsharp 1x4 "windowsPhone8/62×62Logo.scale-140.png"
convert "$base" -resize "44x44" -unsharp 1x4 "windowsPhone8/44×44Logo.scale-100.png"
# Windows phone 8.1 Store Logo
convert "$base" -resize "120x120" -unsharp 1x4 "windowsPhone8/120×120Logo.scale-240.png"
convert "$base" -resize "70x70" -unsharp 1x4 "windowsPhone8/70×70Logo.scale-140.png"
convert "$base" -resize "50x50" -unsharp 1x4 "windowsPhone8/50×50Logo.scale-100.png"
# Windows phone 8.1 Badge Logo
convert "$base" -resize "58x58" -unsharp 1x4 "windowsPhone8/58×58Logo.scale-240.png"
convert "$base" -resize "33x33" -unsharp 1x4 "windowsPhone8/33×33Logo.scale-140.png"
convert "$base" -resize "24x24" -unsharp 1x4 "windowsPhone8/24×24Logo.scale-100.png"


#to support

# App icon (required for all apps)
# 180 x 180
# 120 x 120
# 152 x 152
# 76 x 76
# 167 x 167

# App icon for the App Store (required for all apps)
# 1024 x 1024

#Spotlight search results icon (recommended)
# 180 x 180
# 120 x 120
# 80 x 80
# 60 x 60

# Settings icon (recommended)
# 87 x 87
# 58 x 58
# 29 x 29

# Toolbar and navigation bar icon (optional)
# 66 x 66
# 44 x 44
# 22 x 22

# Web clip icon (recommended for web apps and websites)
# 180 x 180
# 120 x 120
# 152 x 152
# 76 x 76
# 167 x 167


non-square-icons.sh
# Tab bar icon (optional)
# 144 x 96
# 96 x 64
# 96 x 64
# 96 x 64
# 48 x 32
# 96 x 64

# Launch file or image (required for all apps)
# 640 x 1136
# 640 x 960
# 1536 x 2048
# 2048 x 1536
# 768 x 1024
# 1024 x 768
# 2048 x 2732
# 2732 x 2048


# 3.5-Inch Retina Display Screenshots (required) 1min 5max
# 640x990 #portrait
# 960x640 #landscape

# 4-Inch Retina Display Screenshots (Required) 1min 5max
# 640x1136 #hi-res portrait
# 1136x640 #hi-res landscape

# 4.7-inch Retina screenshot  1min 5max
# 750 x 1334 #hi-res portrait
# 1334 x 750 #hi-res landscape


# 5.5-inch Retina screenshot  1min 5max
# 1242 x 2208 #hi-res portrait
# 2208 x 1242 #hi-res landscape

# iPad Screenshots (required if app runs on iPad)   1min 5max
# 1024 x 768 #landscape
# 768 x 1024 #portrait
#
# 2048 x 1536 #hi-res landscape
# 1536 x 2048 #hi-res portrait

#iPad Pro Screenshots   1min 5max
# 2732 x 2048 #hi-res portrait
# 2048 x 2732 #hi-res landscape



#Mac Screenshots OS X only (required) 1min 4max of any of the following sizes
#1280 x 800
#1440 x 900
#2560 x 1600
#2880 x 1800


scripts/screenshots-ios.sh
# This does not do anything yet

# Icon sizes

# 60@2x - 120 x 120
# 60@3x - 140 x 140
# 76 - 76x76
# 76@2x - 152 x 152
# 83.5@2x - 167 x 167
# 1024 - 1024 x 1024
# 2048 - 2048 x 2048


# Screen shot sizes

# iPhone 6 plus, 7 plus, 8 plus
# 5.5-Inch Retina Display
# aspect ratio: 9:16
# 1242 x 2208 or
# 2208 x 1242
#
# iPhone 6, 7, 8
# 4.7-Inch Retina Display
# aspect ratio: 375 : 667
# 750 x 1334 or
# 1334 x 750
#
# iPhone 5, 5s, 5c, sc
# 4-Inch Retina Display
# aspect ratio: 375 : 568
# 640 x 1136 or
# 1136 x 640
#
# iPhone 4, 4s
# 3.5-Inch Retina Display
# aspect ratio: 2 : 3
# 640 x 960 or
# 960 x 640
#
# iPad pro
# 12.9-Inch Retina Display
# Aspect ratio: 512 : 683
# 2048 x 2732 or
# 2732 x 2048
#
# iPad pro, iPad
# 9.7-Inch Retina Display
# Aspect Ratio: 3 : 4
# 2048 x 1536 or
# 1536 x 2048


scripts/icons.sh
#!/bin/sh
base="../input/icon.png"

icons_directory="../output/icons"


web_directory=$icons_directory/web
mkdir -p $web_directory
echo "Creating Web icons in $web_directory"
# Apple touch icons
convert "$base" -resize "57x57" -unsharp 1x4 "$web_directory/icon-57x57.png"
convert "$base" -resize "60x60" -unsharp 1x4 "$web_directory/icon-60x60.png"
convert "$base" -resize "72x72" -unsharp 1x4 "$web_directory/icon-72x72.png"
convert "$base" -resize "76x76" -unsharp 1x4 "$web_directory/icon-76x76.png"
convert "$base" -resize "114x114" -unsharp 1x4 "$web_directory/icon-114x114.png"
convert "$base" -resize "120x120" -unsharp 1x4 "$web_directory/icon-120x120.png"
convert "$base" -resize "144x144" -unsharp 1x4 "$web_directory/icon-144x144.png"
convert "$base" -resize "152x152" -unsharp 1x4 "$web_directory/icon-152x152.png"
convert "$base" -resize "180x180" -unsharp 1x4 "$web_directory/icon-180x180.png"
convert "$base" -resize "192x192" -unsharp 1x4 "$web_directory/apple-icon-precomposed.png"
convert "$base" -resize "192x192" -unsharp 1x4 "$web_directory/apple-icon.png"
# Android touch icon
convert "$base" -resize "192x192"  -unsharp 1x4 "$web_directory/icon-192x192.png"
# favicons
convert "$base" -resize "16x16" -unsharp 1x4 -colors 256 "$web_directory/favicon.ico"
convert "$base" -resize "16x16" -unsharp 1x4 "$web_directory/icon-16x16.png"
convert "$base" -resize "32x32" -unsharp 1x4 "$web_directory/icon-32x32.png"
convert "$base" -resize "96x96" -unsharp 1x4 "$web_directory/icon-96x96.png"
convert "$base" -resize "128x128" -unsharp 1x4 "$web_directory/icon-128x128.png"



ios_directory=$icons_directory/ios
mkdir -p $ios_directory
echo "Creating iOS icons in $ios_directory"
# iPhone App Icon
convert "$base" -resize "120x120" -unsharp 1x4 "$ios_directory/Icon-60@2x.png"
# iPhone App Icon for Retina display
convert "$base" -resize "180x180" -unsharp 1x4 "$ios_directory/Icon-60@3x.png"
# iPad App Icon
convert "$base" -resize "76x76" -unsharp 1x4 "$ios_directory/Icon-76.png"
# iPad App Icon for Retina display
convert "$base" -resize "152x152" -unsharp 1x4 "$ios_directory/Icon-76@2x.png"
# iPad App Icon for Retina display
convert "$base" -resize "167x167" -unsharp 1x4 "$ios_directory/Icon-83.5@2x.png"
# App Icon large
convert "$base" -resize "1024x1024" -unsharp 1x4 "$ios_directory/Icon-1024.png"
# App Submission
convert "$base" -resize "2048x2048" -unsharp 1x4 "$ios_directory/Icon-2048.png"



windowsPhone8_directory=$icons_directory/windowsPhone8
mkdir -p $windowsPhone8_directory
echo "Creating Windows Phone 8 icons in $windowsPhone8_directory"
# Windows phone 8.1 Square 71×71 Logo
convert "$base" -resize "170x170" -unsharp 1x4 "$windowsPhone8_directory/170×170Logo.scale-240.png"
convert "$base" -resize "99x99" -unsharp 1x4 "$windowsPhone8_directory/99×99Logo.scale-140.png"
convert "$base" -resize "71x71" -unsharp 1x4 "$windowsPhone8_directory/71×71Logo.scale-100.png"
# Windows phone 8.1 Square 150×150 Logo
convert "$base" -resize "360x360" -unsharp 1x4 "$windowsPhone8_directory/360×360Logo.scale-240.png"
convert "$base" -resize "210x210" -unsharp 1x4 "$windowsPhone8_directory/210×210Logo.scale-140.png"
convert "$base" -resize "150x150" -unsharp 1x4 "$windowsPhone8_directory/150×150Logo.scale-100.png"
# Windows phone 8.1 Square 44×44 Logo
convert "$base" -resize "106x106" -unsharp 1x4 "$windowsPhone8_directory/106×106Logo.scale-240.png"
convert "$base" -resize "62x62" -unsharp 1x4 "$windowsPhone8_directory/62×62Logo.scale-140.png"
convert "$base" -resize "44x44" -unsharp 1x4 "$windowsPhone8_directory/44×44Logo.scale-100.png"
# Windows phone 8.1 Store Logo
convert "$base" -resize "120x120" -unsharp 1x4 "$windowsPhone8_directory/120×120Logo.scale-240.png"
convert "$base" -resize "70x70" -unsharp 1x4 "$windowsPhone8_directory/70×70Logo.scale-140.png"
convert "$base" -resize "50x50" -unsharp 1x4 "$windowsPhone8_directory/50×50Logo.scale-100.png"
# Windows phone 8.1 Badge Logo
convert "$base" -resize "58x58" -unsharp 1x4 "$windowsPhone8_directory/58×58Logo.scale-240.png"
convert "$base" -resize "33x33" -unsharp 1x4 "$windowsPhone8_directory/33×33Logo.scale-140.png"
convert "$base" -resize "24x24" -unsharp 1x4 "$windowsPhone8_directory/24×24Logo.scale-100.png"
